src/treelearner/feature_histogram.hpp

```diff
@@ -64,6 +64,7 @@ class FeatureHistogram {
       find_best_threshold_fun_ = std::bind(&FeatureHistogram::FindBestThresholdCategorical, this, std::placeholders::_1
         , std::placeholders::_2, std::placeholders::_3, std::placeholders::_4, std::placeholders::_5, std::placeholders::_6);
     }
+    rand_ = Random(meta_->config->seed);
```

Comment:
@btrotta  Please check the following thread: https://github.com/microsoft/LightGBM/pull/2620#discussion_r365257424. I believe `seed` shouldn't be used for anything except generating other seeds.